### PR TITLE
apfs-fuse: init at 20190723

### PR DESCRIPTION
**Pull request by ealasu**
###### Motivation for this change
Add apfs-fuse.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/tools/filesystems/apfs-fuse/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "sgan81";
+    repo = "apfs-fuse";
+    rev    = "309ecb030f38edac4c10fa741a004c5eb7a23e15";
+    sha256 = "0wq6rlqi00m5dp5gbzy65i1plm40j6nsm7938zvfgx5laal4wzr2";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [ fuse3 bzip2 zlib attr cmake ];
```
Comment (symphorien) [inline pkgs/tools/filesystems/apfs-fuse/default.nix]: cmake should be a nativeBuildInput

Comment (ealasu) [reply pkgs/tools/filesystems/apfs-fuse/default.nix]: Thanks, updated.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -1484,6 +1484,11 @@
     github = "eadwu";
     name = "Edmund Wu";
   };
+  ealasu = {
```
Comment (aanderse) [inline maintainers/maintainer-list.nix]: Please have this as a separate commit, before the other.

File: pkgs/tools/filesystems/apfs-fuse/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub, fuse3, bzip2, zlib, attr, cmake }:
+
+stdenv.mkDerivation rec {
+  name = "apfs-fuse-${version}";
```
Comment (aanderse) [inline pkgs/tools/filesystems/apfs-fuse/default.nix]: Please use `pname = "apfs-fuse";` as `name = "${name}-${version}";` is already implied.

File: pkgs/tools/filesystems/apfs-fuse/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  version = "20190723"; # No release tags yet, so just using date of last commit
```
Comment (aanderse) [inline pkgs/tools/filesystems/apfs-fuse/default.nix]: > If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".

File: pkgs/tools/filesystems/apfs-fuse/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    rev    = "309ecb030f38edac4c10fa741a004c5eb7a23e15";
+    sha256 = "0wq6rlqi00m5dp5gbzy65i1plm40j6nsm7938zvfgx5laal4wzr2";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [ fuse3 bzip2 zlib attr ];
+  nativeBuildInputs = [ cmake ];
+
+  meta = with stdenv.lib; {
+    homepage    = https://github.com/sgan81/apfs-fuse;
```
Comment (aanderse) [inline pkgs/tools/filesystems/apfs-fuse/default.nix]: Please quote.

File: pkgs/tools/filesystems/apfs-fuse/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub, fuse3, bzip2, zlib, attr, cmake }:
+
+stdenv.mkDerivation rec {
+  name = "apfs-fuse-${version}";
+  version = "20190723"; # No release tags yet, so just using date of last commit
+
+  src = fetchFromGitHub {
+    owner = "sgan81";
+    repo = "apfs-fuse";
```
Comment (aanderse) [inline pkgs/tools/filesystems/apfs-fuse/default.nix]: A common pattern here is to use `repo = pname;`.

**Review by aanderse — CHANGES_REQUESTED** [review]
Thanks for choosing to contribute this package to nixpkgs. I have left a few comments I hope you find helpful.

File: pkgs/tools/filesystems/apfs-fuse/default.nix (hunk 6 above)
Comment (ealasu) [reply pkgs/tools/filesystems/apfs-fuse/default.nix]: Left it as-is, because I changed `pname` to "apfs-fuse-unstable".

**Comment by ealasu** [comment]
@aanderse Thanks, updated.

**Comment by aanderse** [comment]
@GrahamcOfBorg build apfs-fuse

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -560,6 +560,8 @@ in
 
   antora = callPackage ../development/tools/documentation/antora {};
 
+  apfs-fuse-unstable = callPackage ../tools/filesystems/apfs-fuse/unstable.nix { };
```
Comment (aanderse) [inline pkgs/top-level/all-packages.nix]: @ealasu seeing as there is only a single version of the software you should keep this as `apfs-fuse`.

Comment (ealasu) [reply pkgs/top-level/all-packages.nix]: Done.

**Review by aanderse — APPROVED** [review]

**Comment by aanderse** [comment]
Thanks! :tada:

**Pull request merged**